src/main/java/de/hpi/bpt/argos/storage/DatabaseAccess.java

```diff
@@ -51,11 +52,28 @@
 	 * @param transaction - the current transaction
 	 * @param getValue - the function to get the results from the query
 	 * @param defaultValue - a fall back default value in case anything went wrong or no entities were found
-	 * @param <R> - the result type
-	 * @param <Q> - the query type
+	 * @param <ResultType> - the result type
+	 * @param <QueryType> - the query type
 	 * @return - an object of the result value type
 	 */
-	<R, Q> R getArtifacts(Session session, Query<Q> query, Transaction transaction, Callable<R> getValue, R defaultValue);
+	<ResultType, QueryType> ResultType getArtifacts(Session session,
+													Query<QueryType> query,
+													Transaction transaction,
+													Callable<ResultType> getValue,
+													ResultType defaultValue);
+
+	/**
+	 * This method returns a list of entities, which will get fetched by their ids.
+	 * @param <ResultType> - the result type
+	 * @param <ResultImplType> - the implementation type of the result type
+	 * @param session - the database session, which must be open
+	 * @param resultTypeClass - the class of the result type
+	 * @param ids - the ids to fetch from the database
+	 * @return - a list of all entities, which were contained in the ids
+	 */
+	<ResultType, ResultImplType extends ResultType> List<ResultType> getArtifactsById(Session session,
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this generic name to match the regular expression '^[A-Z][0-9]?$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS00119)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this generic name to match the regular expression '^[A-Z][0-9]?$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS00119)